### PR TITLE
refactor: followup of remove -deprecatedrpc=addresses flag

### PR DESCRIPTION
**Pull request by fanquake**
I've cherry-picked some of the commits out of #22924, and made minor changes (like fixing named args).

### CODE REVIEW

**Review by maflcko — APPROVED** [review]
review ACK 9aad2844aa7047121e72cb0bd416cc14101e2b7c 😼

<details><summary>Show signature</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

review ACK 9aad2844aa7047121e72cb0bd416cc14101e2b7c 😼
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUigxAwAuyqNsS2+EPh36vJ7QjbJzWcX1/55TU4yjULBwisAqfcoU06nfp5nmlWB
lJ7y4iJJ/ZgcS/XP5Lf3yCKz6KmnViT1T6CGrEVsIFnZQ70cyzMm+Ikt2NkeDZE5
f8QqOW6wucqWC4kt8Cu9557cuwYQy7lkk7fZdNcOQhN8PC0J2MKp6VOf3Q63cQYF
HJmM5cXeo0e5+O+k+aAKABxOEjV5dJH8dKXZfPVgb8jeZuHlm1PTvRsIulWYWPRg
WbZXkRcN/+v0EEetVa8IhYV4tWpoiWzHXrGLnTo4MzDttsL7PFVKba3Pv01iXflx
eY+wbE6W4wwYfTn9S+UwHFA5MolNeuwMnC2N1MWuI6HBMqQ2SilP1zbb5bX2QZby
VLiJ4UB8pMDWqY2mBsDEfGKscgdOR6qK38zH8iFmfWIwKphFT5sRzTHR2f3rALmN
u3NnyMcmn+YvWyDgyTWC/e2g2TQeIAHeux9wQHrcyAXqvOw9qtTab5EE1FexLk/T
JdbYgGFz
=/ATi
-----END PGP SIGNATURE-----
```


</details>

File: src/test/fuzz/script_format.cpp [hunk 1]
```diff
@@ -29,7 +29,7 @@ FUZZ_TARGET_INIT(script_format, initialize_script_format)
     (void)ScriptToAsmStr(script, /*fAttemptSighashDecode=*/fuzzed_data_provider.ConsumeBool());
 
     UniValue o1(UniValue::VOBJ);
-    ScriptPubKeyToUniv(script, o1, /*include_hex=*/fuzzed_data_provider.ConsumeBool());
+    ScriptToUniv(script, o1, /*include_hex=*/fuzzed_data_provider.ConsumeBool(), /*include_address=*/fuzzed_data_provider.ConsumeBool());
     UniValue o3(UniValue::VOBJ);
     ScriptToUniv(script, o3);
```
[the remark this answers is not in the document]
Comment (fanquake) [reply src/test/fuzz/script_format.cpp]: Removed

File: src/core_write.cpp [hunk 2]
```diff
@@ -253,8 +253,8 @@ void TxToUniv(const CTransaction& tx, const uint256& hashBlock, UniValue& entry,
         entry.pushKV("fee", ValueFromAmount(fee));
     }
 
-    if (!hashBlock.IsNull())
-        entry.pushKV("blockhash", hashBlock.GetHex());
+    if (!block_hash.IsNull())
```
[the remark this answers is not in the document]
Comment (fanquake) [reply src/core_write.cpp]: Added

File: src/rest.cpp [hunk 3]
```diff
@@ -855,7 +855,7 @@ static bool rest_getutxos(const std::any& context, HTTPRequest* req, const std::
 
             // include the script in a json output
             UniValue o(UniValue::VOBJ);
-            ScriptPubKeyToUniv(coin.out.scriptPubKey, o, true);
+            ScriptToUniv(coin.out.scriptPubKey, o, true, /*include_address=*/true);
```
[the remark this answers is not in the document]
Comment (fanquake) [reply src/rest.cpp]: Should be addressed

**Comment by MarcoFalke** [comment]
re-ACK 9563a645c22a455da3d2d305ed0eef4266b1d322 🕓

<details><summary>Show signature</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

re-ACK 9563a645c22a455da3d2d305ed0eef4266b1d322 🕓
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUjyqQv/WChwsfIm2X3iwc5IJ6TYwiH81nmMk6ELbpIwzjD3uqWS3ordwelmW9jH
MbCxAv44LJ4IBG6XzFXKBKwuHCmuky7/5EfmBVlX/H6ztnamQMXDbYswpYpQiIq+
HXu8f2pDe/+U025h36rALlmc9OuCBQWlKmBzktnOhCaSg4LD+72qqQQ+B8Avr6DE
1tv/s0u1/5lC8DvsjENGJi+ah6fO87qp2n5+deFFEBpQt40EmHbVJcHM+suDvP3d
qIBomRIYP5OxGB1oQ4xcq5+7/qpgdxh/elN9Q1diaDWuCmYQ46Omnft8yvWu4YSS
0fLoFkXztAMYgAd4yrgs8OUQAN4DdSc3xBuYwtaxmhHvFl+wnWUhScFE7/bTtjRb
RlrEljbFIYjJASQzETp9WHhgfMRq6ARaJc+ulS0+g1UAMNg3PX8taSkX7YmJaN2l
Q4JO2N/Rw0feV94pOKNyOByhkwieAz7izabBeahq/Yr+2iJzGKSrxzOUar5z9G0K
KjIqtwNr
=O5zM
-----END PGP SIGNATURE-----
```


</details>

**Comment by mjdietzx** [comment]
Post merge review ACK 9563a645c22a455da3d2d305ed0eef4266b1d322

**Pull request merged**